service/src/main/java/com/fecklessweasel/service/UserSessionUtil.java

```diff
@@ -0,0 +1,173 @@
+ * @author Christian Gunderman
+ */
+public final class UserSessionUtil {
+    /** The client side cookie for storing user authentication token. */
+    private static final String AUTH_COOKIE = "FecklessWeaselAuthToken";
+
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private UserSessionUtil() { }
+
+    /**
+     * Starts a new session and logs in the user.
+     * @param response The HTTP response for the HTTP request. Method
+     * automatically sets the authentication cookie. Caller must sendRedirect
+     * to desired end UI page.
+     * @param username The username of the user to log in.
+     * @param password The password of the user to log in.
+     * @throws ServiceException Thrown if the service encounters an error.
+     */
+    public static UserSession startSession(Connection connection,
+                                           HttpServletResponse response,
+                                           String username,
+                                           String password) throws ServiceException {
+
+        // Start the session.
+        UserSession session = UserSession.start(connection,
+                                                username,
+                                                password);
+
+        // Store session in a cookie.
+        Cookie authCookie = new Cookie(AUTH_COOKIE,
+                                       session.getSessionString());
+
+        // Make this cookie available anywhere on site hostname.
+        authCookie.setPath("/");
+
+        // TODO: uncomment when SSL is enabled.
+        // Prevent cookie from being sent over non secure protocol (no HTTP, only HTTPS).
+        // authCookie.setSecure(true);        
+        
+        response.addCookie(authCookie);
+
+        return session;
+    }
+
+    /**
+     * Resumes a session from the authentication cookie stored in a request.
+     * @param connection The connection to the database. Provided by SQLSource.
+     * @param request The HTTP request containing the authentication cookie.
+     * @throws ServiceException Thrown if the service encounters an exception.
+     * INVALID_SESSION_HEADER, ACCESS_DENIED, or NOT_AUTHENTICATED ServerStatus
+     * if unable to authenticate.
+     * @return The resumed UserSession.
+     */
+    public static UserSession resumeSession(Connection connection,
+                                            HttpServletRequest request)
+        throws ServiceException {
+
+        Cookie[] cookies = request.getCookies();
+
+        if (cookies == null) {
+            return null;
+        }
+        
+        // Run through all cookies for our domain. This is ok as long as the session
+        // cookie is our only cookie. If we start saving more than five or ten
+        // we should use a different strategy for indexing and searching.
+        for (Cookie cookie : cookies) {
+            if (cookie.getName().equals(AUTH_COOKIE)) {
```

Comment:
Yes they can, ideally they should be encapsulated in a single class though that accepts the request and response and writes and reads them in one place.